IGSIOCalibration/vtkIGSIOAbstractCalibrationAlgo.h

```diff
@@ -0,0 +1,224 @@
+/*=IGSIO=header=begin======================================================
+Program: IGSIO
+Copyright (c) Laboratory for Percutaneous Surgery. All rights reserved.
+See License.txt for details.
+=========================================================IGSIO=header=end*/
+
+#ifndef __vtkIGSIOAbstractCalibrationAlgo_h
+#define __vtkIGSIOAbstractCalibrationAlgo_h
+
+// Local includes
+#include "igsioConfigure.h"
+#include "igsioCommon.h"
+#include "vtkIGSIOCalibrationExport.h"
+
+// VTK includes
+#include <vtkCommand.h>
+#include <vtkObject.h>
+#include <vtkMatrix4x4.h>
+
+// STL includes
+#include <set>
+#include <deque>
+
+class vtkXMLDataElement;
+
+//-----------------------------------------------------------------------------
+
+/*!
+  \class vtkIGSIOAbstractCalibrationAlgo
+  \brief Abstract calibration algorithm
+
+  Abstract class for performing calibration algorithms.
+
+  Input transforms are collected and stored in an input buffer that inserts each transform into a bucket.
+  New buckets are added as each are filled, up to a specified maximum. If a bucket is filled, then it is checked
+  to ensure it contains valid calibration poses, and the error within the bucket is below a specified threshold.
+  If the poses are invalid or the error is too high, then the entire buffer is discarded.
+
+  \ingroup igsioCalibrationAlgorithm
+*/
+class vtkIGSIOCalibrationExport vtkIGSIOAbstractCalibrationAlgo : public vtkObject
+{
+public:
+  vtkTypeMacro(vtkIGSIOAbstractCalibrationAlgo, vtkObject);
+
+  enum Events
+  {
+    InputTransformAddedEvent = vtkCommand::UserEvent + 173,
+  };
+
+  enum CalibrationErrorCodes
+  {
+    CALIBRATION_FAIL = IGSIO_FAIL,
+    CALIBRATION_SUCCESS = IGSIO_SUCCESS,
```

Comment:
It is a bit unusual to have success code == 1 and everything else failure. Maybe we could have a result (FAIL/SUCCESS) flag and a separate error code enum (and/or error string) that gives details about the current state.

Reply:
I've updated the enum so that there is a NO_ERROR enum, rather than SUCCESS.
The success/failure of the final calibration is indicated by the return value of Do(Pivot/Spin)Calibration.